libraries/chain/database.cpp

```diff
@@ -2252,15 +2252,93 @@ namespace golos { namespace chain {
             try {
                 uint128_t total_weight(c.total_vote_weight);
                 //edump( (total_weight)(max_rewards) );
+                uint128_t votes_in_auction_window_weight;
+                uint128_t votes_after_auction_window_weight;
+                
                 share_type unclaimed_rewards = max_rewards;
 
+                share_type votes_in_auction_window_reward;
+                share_type votes_after_auction_window_reward;
+                share_type auction_window_reward;
+
+                auto auw_time = c.created + c.auction_window_size;
+
                 if (c.total_vote_weight > 0 && c.allow_curation_rewards) {
+                    // If auction window reward is chosen to go to curators, we need to
+                    // separated votes for 2 sets: 
+                    // c.created     auction_window               cashout
+                    // |_______________|____________________________|
+                    //     1                      2 
+                    // 1st set contains votes with last_update from [c.created, c.created + auction_window_size)
+                    // 2st set contains votes with last_update from [c.created + auction_window_size, cashout]
+
+                    // Also we need to distibute auction_window reward to voters from second set.
+                    // So we need to calculate parts of reward for 3 sets:
+                    //       - votes_in_auction_window_reward
+                    //       - votes_after_auction_window_reward
+                    //       - auction_window_reward 
+
+                    if (has_hardfork(STEEMIT_HARDFORK_0_19__898) && 
+                        c.auction_window_reward_destination == protocol::to_curators
+                    ) {
+                        // separate votes
+                        const auto &cvlupdidx = get_index<comment_vote_index>().indices().get<by_vote_last_update>();
+                        auto itr = cvlupdidx.lower_bound(boost::make_tuple(0, c.created));
+                        auto itr_after_auw = cvlupdidx.lower_bound( // auw -- auctcion window 
+                            boost::make_tuple(0, c.created + c.auction_window_size)
+                        );
+
+                        while (itr != itr_after_auw && itr->comment == c.id) {
```

Comment:
You shouldn't do such iterations on runtime - it's too heavy and will generate forks.